# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1219244484 14400
#      Wed Aug 20 11:01:24 2008 -0400
# Branch release-3-0-x
# Node ID 018dff0b30d2ad73843649cf046ff47eac39b2cd
# Parent  7212edd02db13c3a14b4c7fb263106edb9ad02c0
avoid problematic subfunction call in an anonymous function

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,14 +1,19 @@
 2008-07-28  David Bateman  <dbateman@free.fr>
  
 	* plot/contourf.m: Allow X,Y to be unequal vectors.
 
 	* plot/shading.m: Clarify help string.
 	
+2008-08-20  Jaroslav Hajek <highegg@gmail.com>
+
+	* pkg/pkg.m (configure_make): Pass handle to is_architecture_dependent
+	directly.
+
 2008-08-20  David Bateman  <dbateman@free.fr>
 
 	* plot/__go_draw_axes__.m: Don't set pm3d implicit if the plot
 	contains a surface. Fixes things like meshc(peaks()).
 
 2008-08-19  Jaroslav Hajek <highegg@gmail.com>
 
 	* optimization/glpk.m: Fix invalid call to zeros.
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -1297,17 +1297,17 @@ function configure_make (desc, packdir, 
       endif
       filenames = split_by (filenames, " ");
     endif
 
     ## Split into architecture dependent and independent files
     if (isempty (filenames))
       idx = [];
     else
-      idx = cellfun (@(x) is_architecture_dependent (x), filenames);
+      idx = cellfun (@is_architecture_dependent, filenames);
     endif
     archdependent = filenames (idx);
     archindependent = filenames (!idx);
 
     ## Copy the files
     if (! all (isspace ([filenames{:}])))
 	if (! exist (instdir, "dir")) 
 	  mkdir (instdir);
